### ISSUE TITLE
Error creating lock: endpoints lock is removed, migrate to endpointsleases

### ISSUE THREAD

**Issue by rajch** [kubernetes-sigs/sig-storage-lib-external-provisioner#123]
On migrating from v6 to v8, my provisioner crashes with the following error:

```
F0720 16:59:52.441682    5606 controller.go:878] Error creating lock: endpoints lock is removed, migrate to endpointsleases
```

I can suppress it by disabling leader election, but would like to know how to solve the problem.

**Comment by rajch** [comment]
Found the problem. In the v8.0.0 tagged release, line 868 calls `resourcelock.New` with a hardcoded locktype of "endpoints", which is no longer supported. Here are the problem lines, in:
*// https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/blob/v8.0.0/controller/controller.go*
```
	if ctrl.leaderElection {
		rl, err := resourcelock.New("endpoints",
			ctrl.leaderElectionNamespace,
			strings.Replace(ctrl.provisionerName, "/", "-", -1),
			ctrl.client.CoreV1(),
			nil,
			resourcelock.ResourceLockConfig{
				Identity:      ctrl.id,
				EventRecorder: ctrl.eventRecorder,
			})
		if err != nil {
			klog.Fatalf("Error creating lock: %v", err)
		}
```

This has been corrected in HEAD of the master branch, but a new release has not been made with the correction.

**Comment by rajch** [comment]
Could we have a new release, please?

**Comment by roi-codefresh** [comment]
Will this be released officially anytime soon?

**Comment by jsafrane** [comment]
I've just released v9.0.0. Sorry about the long wait.

**Comment by rajch** [comment]
Thank you!
